src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/SafetyServiceImpl.java

```diff
@@ -0,0 +1,233 @@
+        parentIDs.stream().forEach(parentID -> {
+            EntitySafetyWSHub cloned = new EntitySafetyWSHub(entitySafetywshub);
+            cloned.setEntity(entity);
+            cloned.setEntitySafetyCode(parentID);
+            safetiesToAdd.add(cloned);
+        });
+
+        return tools.batchOperation(context, this::addSafety, safetiesToAdd);
+    }
+
+    /**
+     * Deletes a safety given its id (safety code).
+     *
+     * @param context the user credentials
+     * @param safetyCode the safety code of the safety to delete
+     * @return
+     * @throws InforException
+     */
+    @Override
+    public String deleteSafety(InforContext context, String safetyCode) throws InforException {
+        SafetyService.validateInput(safetyCode);
+
+        MP3221_DeleteEntitySafety_001 deleteEntitySafety = new MP3221_DeleteEntitySafety_001();
+        deleteEntitySafety.setSAFETYCODE(safetyCode);
+
+        tools.performInforOperation(context, inforws::deleteEntitySafetyOp, deleteEntitySafety);
+        return "OK";
+    }
+
+    /**
+     * Deletes safeties given their ids (safety codes).
+     *
+     * @param context the user credentials
+     * @param safetyCodes a list of the safety codes of the safeties to delete
+     * @return
+     * @throws InforException
+     */
+    @Override
+    public BatchResponse<String> deleteSafetiesBatch(InforContext context, List<String> safetyCodes) throws InforException {
+        // Validate not needed now, since performed at .deleteSafety
+
+        return tools.batchOperation(context, this::deleteSafety, safetyCodes);
+    }
+
+    /**
+     * Updates the specified safety with the given values of the safety object.
+     *
+     * @param context the user credentials
+     * @param entitySafetywshub the safety to be updated, with its updated values
+     * @return the updated safety object
+     * @throws InforException
+     */
+    @Override
+    public String syncEntitySafety(InforContext context, EntitySafetyWSHub entitySafetywshub) throws InforException {
+        SafetyService.validateInput(entitySafetywshub);
+
+        MP3222_GetEntitySafety_001 getEntitySafety = new MP3222_GetEntitySafety_001();
+        getEntitySafety.setSAFETYCODE(entitySafetywshub.getSafetyCode());
+        MP3222_GetEntitySafety_001_Result resGet = tools.performInforOperation(context, inforws::getEntitySafetyOp, getEntitySafety);
+        EntitySafety entitySafetyInfor = resGet.getResultData().getEntitySafety();
+
+        tools.getInforFieldTools().transformWSHubObject(entitySafetyInfor, entitySafetywshub, context);
+
+        MP3220_SyncEntitySafety_001 syncEntitySafety = new MP3220_SyncEntitySafety_001();
+
+        syncEntitySafety.setEntitySafety(entitySafetyInfor);
+        MP3220_SyncEntitySafety_001_Result resSync = tools.performInforOperation(context, inforws::syncEntitySafetyOp, syncEntitySafety);
+
+        return resSync.getResultData().getSAFETYCODE();
+    }
+
```

Comment:
Maybe one new line more to make it clear there is nothing here?

Reply:
Maybe a comment "//Nothing here"?